### ISSUE TITLE
false sc2030

### ISSUE THREAD

**Issue by pan64** [koalaman/shellcheck#1585]
#### For bugs
- Rule Id (if any, e.g. SC1000):  SC2030, SC2031
- My shellcheck version (`shellcheck --version` or "online"): 0.4.7 and online
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit

#### For new checks and feature suggestions
- [ ] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [ ] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/bin/bash

function process_repo() {
        :
}

function search_repos() {
    local failed=0
    while read -r line; do
        repo="$line"
        echo "processing ${repo}" >&3
        echo "${repo}"
        process_repo "${repo}" || failed=1
    done < "${ACTIVE_REPOS}"               # << line 14
    return "${failed}"
}


```

#### Here's what shellcheck currently says:
this is ok and reported ok, but if I modify line 14: add a | (pipe):
```sh
 done < "${ACTIVE_REPOS}"  | sort -u
```
shellcheck will report SC2030 and SC2031 on variable failed, which is obviously false

#### Here's what I wanted or expected to see:
do not mix double and single pipes (or ??).

**Comment by koalaman** [comment]
ShellCheck is right, this is not ok. Here's a simplified example:

```
#!/bin/bash

failed=0
while read -r _; do
  echo "FAILED"
  failed=1
done < <(echo hello) | sort -u
echo "Failed: $failed"
```

Due to the pipeline, the assignment to `failed` is lost:

```
FAILED
Failed: 0
```

if you remove the `| sort -u`, you get 1 as expected.

**Comment by pan64** [comment]
quite interesting. thanks
